ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/PruningITSpec.scala

```diff
@@ -0,0 +1,162 @@
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.client.binding
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{AppendedClues, Checkpoints, EitherValues, OptionValues}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class PruningITSpec
+    extends AsyncFlatSpec
+    with BenchtoolSandboxFixture
+    with SuiteResourceManagementAroundAll
+    with Matchers
+    with AppendedClues
+    with EitherValues
+    with OptionValues
+    with Checkpoints {
+
+  private var actorSystem: ActorSystem[SpawnProtocol.Command] = _
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    actorSystem = ActorSystem(Creator(), "Creator")
+  }
+
+  override protected def afterAll(): Unit = {
+    super.afterAll()
+    actorSystem.terminate()
+  }
+
+  it should "populate participant with contracts using specified application-ids and submitters" in {
+    val submissionConfig = WorkflowConfig.FooSubmissionConfig(
+      numberOfInstances = 100,
+      numberOfObservers = 1,
+      uniqueParties = false,
+      instanceDistribution = List(
+        WorkflowConfig.FooSubmissionConfig.ContractDescription(
+          template = "Foo1",
+          weight = 1,
+          payloadSizeBytes = 0,
+        )
+      ),
+      consumingExercises = Some(
+        WorkflowConfig.FooSubmissionConfig.ConsumingExercises(
+          probability = 0.5,
+          payloadSizeBytes = 0,
+        )
+      ),
+    )
+    val postPruneSubmissionConfig = WorkflowConfig.FooSubmissionConfig(
+      numberOfInstances = 1,
+      numberOfObservers = 1,
+      uniqueParties = false,
+      instanceDistribution = List(
+        WorkflowConfig.FooSubmissionConfig.ContractDescription(
+          template = "Foo2",
+          weight = 1,
+          payloadSizeBytes = 0,
+        )
+      ),
+    )
+    val testedPruningBenchmark =
+      new PruningBenchmark(reportingPeriod = Config.Default.reportingPeriod)
+    for {
+      (apiServices, allocatedParties, submission) <- benchtoolFooSubmissionFixture(submissionConfig)
+      _ <- submission.performSubmission(submissionConfig)
+      txPrePruning: ObservedEvents <- txTreeObserver(
+        apiServices = apiServices,
+        party = allocatedParties.observers(0),
+      )
+      pruningBenchmarkResult <- testedPruningBenchmark.benchmarkPruning(
+        pruningConfig = WorkflowConfig.PruningConfig(
+          name = "pruning-benchmark-test",
+          pruneAllDivulgedContracts = true,
+          maxDurationObjective = 0.nano,
+        ),
+        regularUserServices = apiServices,
+        adminServices = apiServices,
+        actorSystem = actorSystem,
+      )
+      (apiServices, allocatedParties, submission) <- benchtoolFooSubmissionFixture(
+        postPruneSubmissionConfig
+      )
+      _ <- submission.performSubmission(postPruneSubmissionConfig)
```

Comment:
What purpose does this serve ?

Reply:
Ah, that's a left over of when I tried to use a tx stream to see the effects of the pruning. Removing